downloader/src/main/kotlin/VcsHost.kt

```diff
@@ -44,6 +46,52 @@ enum class VcsHost(
      */
     vararg supportedTypes: VcsType
 ) {
+    AZURE_DEVOPS("dev.azure.com", VcsType.GIT) {
+        private val commitPrefix = "GC"
+
+        override fun getUserOrOrgInternal(projectUrl: URI) = projectUrlToUserOrOrgAndProject(projectUrl)?.first
+
+        override fun getProjectInternal(projectUrl: URI) = projectUrl.path.substringAfterLast("/")
+
+        override fun toVcsInfoInternal(projectUrl: URI): VcsInfo {
+            val url = projectUrl.toString().replace(projectUrl.query.toString(), "").dropLast(1)
+            val revision =
+                projectUrl.getQueryParameterValue("version").firstOrNull()?.withoutPrefix(commitPrefix).orEmpty()
```

Comment:
Personally, I'd rather wrap this at the end of the line, grouping chained calls that somewhat belong together, like

            val revision = projectUrl.getQueryParameterValue("version").firstOrNull()
                ?.withoutPrefix(commitPrefix).orEmpty()